test/tomography/test_gateset_tomography.py

```diff
@@ -0,0 +1,124 @@
+                                noise_model=None,
+                                gateset_basis='Standard GST'):
+        backend_qasm = Aer.get_backend('qasm_simulator')
+        circuits = gateset_tomography_circuits(gateset_basis=gateset_basis)
+        qobj = assemble(circuits, shots=shots)
+        result = backend_qasm.run(qobj, noise_model=noise_model).result()
+        fitter = GatesetTomographyFitter(result, circuits, gateset_basis)
+        return fitter
+
+    @staticmethod
+    def expected_linear_inversion_gates(Gs, Fs):
+        rho = np.array([[0.5], [0], [0], [0.5]])
+        B = np.array([(F @ rho).T[0] for F in Fs]).T
+        return {label: np.linalg.inv(B) @ G @ B for (label, G) in Gs.items()}
+
+    @staticmethod
+    def hs_distance(A, B):
+        return sum([np.abs(x) ** 2 for x in np.nditer(A-B)])
+
+    def compare_gates(self, expected_gates, result_gates, labels, delta=0.2):
+        for label in labels:
+            expected_gate = expected_gates[label]
+            result_gate = result_gates[label]
+            msg = "Failure on gate {}: Expected gate = \n{}\n" \
+                  "vs Actual gate = \n{}".format(label,
+                                                 expected_gate,
+                                                 result_gate)
+            distance = self.hs_distance(expected_gate, result_gate)
+            self.assertAlmostEqual(distance, 0, delta=delta, msg=msg)
+
+    def run_test_on_basis_and_noise(self,
+                                    gateset_basis=StandardGatesetBasis,
+                                    noise_model=None,
+                                    noise_ptm=None):
+        labels = gateset_basis.gate_labels
+        gates = gateset_basis.gate_matrices
+
+        # apply noise if given
+        for label in labels:
+            if label != "Id" and noise_ptm is not None:
+                gates[label] = noise_ptm @ gates[label]
+        Fs = [gateset_basis.spam_matrix(label)
+              for label in gateset_basis.spam_labels]
+
+        # prepare the fitter
+        fitter = self.collect_tomography_data(shots=10000,
+                                              noise_model=noise_model,
+                                              gateset_basis=gateset_basis)
+
+        # linear inversion test
+        expected_gates = self.expected_linear_inversion_gates(gates, Fs)
+        result_gates = fitter.linear_inversion()
+        self.compare_gates(expected_gates, result_gates, labels)
+
+        # fitter optimization test
+        expected_gates = gates
+        result_gates = fitter.fit()
+        self.compare_gates(expected_gates, result_gates, labels)
+
+    def test_noiseless_standard_basis(self):
+        self.run_test_on_basis_and_noise()
+
+    def test_amplitude_damping_standard_basis(self):
+        gamma = 0.05
+        A0 = [[1, 0], [0, np.sqrt(1 - gamma)]]
+        A1 = [[0, np.sqrt(gamma)], [0, 0]]
+        noise_ptm = np.array([[1, 0, 0, 0],
```

Comment:
I would make this a `PTM` object

Reply:
I considered it but we don't actually use anything this (generally useful) class has to offer. I could use the PTM class to compute the matrix directly from a Kraus representation of the noise, but I preferred writing it explicitly.

Reply:
You don't need to compute from a Kraus, you can just do `PTM(ptm_array)` and it will be automatically converted when adding to a noise model for Aer simulation.

For example this should work:
```python
def run_test_on_basis_and_noise(self,
                                gateset_basis=StandardGatesetBasis,
                                noise_model=None,
                                noise_ptm=None):
    # ...

    # apply noise if given
    for label in labels:
        if label != "Id" and noise_ptm is not None:
            # Wouldn't need .data here if updated to use
            # QuantumChannel objects in the rest of the code
            gates[label] = noise_ptm.data @ gates[label]
    # ...
    
def test_amplitude_damping_standard_basis(self):
    gamma = 0.05
    noise_ptm = PTM(np.array([[1, 0, 0, 0],
                              [0, np.sqrt(1-gamma), 0, 0],
                              [0, 0, np.sqrt(1-gamma), 0],
                              [gamma, 0, 0, 1-gamma]]))
    noise_model = NoiseModel()
    noise_model.add_all_qubit_quantum_error(noise_ptm, ['u1', 'u2', 'u3'])
    self.run_test_on_basis_and_noise(noise_model=noise_model,
                                     noise_ptm=noise_ptm)

def test_depolarization_standard_basis(self):
    p = 0.05
    noise_ptm = PTM(np.array([[1, 0, 0, 0],
                              [0, 1-p, 0, 0],
                              [0, 0, 1-p, 0],
                              [0, 0, 0, 1-p]]))
    noise = depolarizing_error(p, 1)  # 1-qubit error
    noise_model = NoiseModel()
    noise_model.add_all_qubit_quantum_error(noise_ptm, ['u1', 'u2', 'u3'])
    self.run_test_on_basis_and_noise(noise_model=noise_model,
                                     noise_ptm=noise_ptm)
```

Similarly you could define the example channels using `Kraus` objects and directly add them to Aer noise model the same way, and automatically convert them to PTM if you need them in that form using

```python
def test_amplitude_damping_standard_basis(self):
    gamma = 0.05
    A0 = [[1, 0], [0, np.sqrt(1 - gamma)]]
    A1 = [[0, np.sqrt(gamma)], [0, 0]]
    noise = Kraus([A0, A1])
    noise_model = NoiseModel()
    noise_model.add_all_qubit_quantum_error(noise, ['u1', 'u2', 'u3'])
    # Note that the following PTM conversion would be better done inside
    # the `run_test_on_basis_and_noise` function
    self.run_test_on_basis_and_noise(noise_model=noise_model,
                                         noise_ptm=PTM(noise)). 
```

Reply:
Done